src/Makefile

```diff
@@ -9,33 +9,22 @@ CFLAGS += $(GNUTLS_CFLAGS) $(P11KIT_CFLAGS) $(BDB_CFLAGS) $(TASN1_CFLAGS) $(UNBO
 
 LDFLAGS += -std=gnu11
 
-LIBS = $(GNUTLS_LIBS) $(P11KIT_LIBS) $(BDB_LIBS) $(TASN1_LIBS) $(UNBOUND_LIBS) $(QUICKDER_LIBS)
-LIBS += -lldns -lpthread
+LIBS = $(GNUTLS_LIBS) $(P11KIT_LIBS) $(BDB_LIBS) $(TASN1_LIBS) $(UNBOUND_LIBS) $(QUICKDER_LIBS) 
+LIBS += $(LDNS_LIBS) -lpthread
 
 ifdef WITH_SYSTEMD
 CFLAGS += -DHAVE_SYSTEMD
-LIBS   += -lsystemd-daemon
+LIBS   += $(SYSTEMD_LIBS) 
 endif
```

Comment:
Basically the same -- you don't need this for building on NixOS afaik, so I feel reluctant to change.